### PR TITLE
Feat(Task-1): As a developer, I want a solid start point for the Weather Project

### PR DESCRIPTION
**Pull request by SamuFigueirido**

### CODE REVIEW

**Review by diegopf — CHANGES_REQUESTED** [review]
As starting point, do not commit your node_modules into your repository :) Add the folder to the .gitignore and remove them from the branch

**Review by eloydotco — CHANGES_REQUESTED** [review]
Add /node_modules to .gitignore file

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules/
```
Comment (diegopf) [inline .gitignore]: You should also add dist folder here

File: config/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+const path = require('path');
+module.exports = {
+    entry: './src/app/app.module.js',
+    output: {
+        filename: 'bundle.js',
+        path: path.resolve(__dirname, '../dist')
+    },
+    devServer: {
```
Comment (diegopf) [inline config/webpack.config.js]: You should split the specific environment configuration between different files

File: package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "CaelumWeather",
+  "version": "1.0.0",
+  "description": "World weather application",
+  "main": "app.module.js",
+  "private": true,
+  "scripts": {
+    "dev": "webpack --mode development --config ./config/webpack.config.js",
+    "build": "webpack --mode production --config ./config/webpack.config.js"
```
Comment (diegopf) [inline package.json:9]: You should make use of HtmlWebpackPlugin in order to produce the index.html to serve your bundle

**Review by diegopf — CHANGES_REQUESTED** [review]

File: config/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+var HtmlWebpackPlugin = require('html-webpack-plugin');
```
Comment (eloydotco) [inline config/webpack.config.js]: Use const/let instead of var

File: src/app/app.module.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+import angular from 'angular';
+import appComponent from './app.component';
+angular.module('app', [
```
Comment (eloydotco) [inline src/app/app.module.js]: why this line break??

File: src/index.html [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<body ng-app="app" ng-strict-di ng-cloak>
```
Comment (eloydotco) [inline src/index.html:9]: I think some html tags are missing here...

**Review by eloydotco — CHANGES_REQUESTED** [review]

File: config/webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+const path = require('path');
+module.exports = {
+    entry: './src/app/app.module.js',
+    output: {
+        filename: 'bundle.js',
+        path: path.resolve(__dirname, '../dist')
+    },
+    plugins: [new HtmlWebpackPlugin()],
```
Comment (gbtec-ext-jorgevillar) [inline config/webpack.config.js]: I think that you need to configure the pluging, at least adding the filename(resulting html) and template(your main html route) options

**Review by gbtec-ext-jorgevillar — CHANGES_REQUESTED** [review]

**Review by eloydotco — APPROVED** [review]

**Review by diegopf — APPROVED** [review]

**Review by gbtec-ext-jorgevillar — APPROVED** [review]

**Pull request merged**